libvast/vast/data.hpp

```diff
@@ -158,6 +158,19 @@ class data : detail::totally_ordered<data>,
   friend bool operator==(const data& lhs, const data& rhs);
   friend bool operator<(const data& lhs, const data& rhs);
 
+  // These operators need to be templates so they're instantiated at a later
+  // point in time, because there'd be a cyclic dependency otherwise.
+  // caf::variant<Ts...> is just a placeholder for vast::data_view here.
+  template <typename... Ts>
+  friend bool operator==(const data& lhs, const caf::variant<Ts...>& rhs) {
+    return is_equal(lhs, rhs);
+  }
+
+  template <typename... Ts>
+  friend bool operator==(const caf::variant<Ts...>& lhs, const data& rhs) {
+    return is_equal(lhs, rhs);
+  }
+
```

Comment:
We should probably update this comment in `vast/view.hpp` at the same time:

```
// TODO: we cannot use operator == and != here because data has a non-explicit
// constructor, which results in error all over the code base. Therefore, we
// work around this by giving this function a name.

bool is_equal(const data& x, const data_view& y);

bool is_equal(const data_view& x, const data& y);
```